src/ng/directive/ngOptions.js

```diff
@@ -325,12 +325,28 @@ var ngOptionsDirective = ['$compile', '$parse', function($compile, $parse) {
         // The option values were already computed in the `getWatchables` fn,
         // which must have been called to trigger `getOptions`
         var optionValues = valuesFn(scope) || [];
+        var optionValuesKeys;
 
-        var keys = Object.keys(optionValues);
-        keys.forEach(function getOption(key) {
+
+        if (!keyName && isArrayLike(optionValues)) {
```

Comment:
It doesn't prevent it from working, it is saying that if you explicitly are iterating over an object's keys (via `(key,value) in obj`) then you don't want `ngOptions` to treat it like an array.